src/repo/repo_test.go

```diff
@@ -0,0 +1,94 @@
+package repo
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing/object"
+)
+
+func TestClone(t *testing.T) {
+	tempDir, err := os.MkdirTemp("", "test-repo-clone")
+	if err != nil {
+		t.Fatalf("Failed to create temp dir: %v", err)
+	}
+	defer os.RemoveAll(tempDir)
+
+	mockRepoPath := filepath.Join(tempDir, "mock-repo")
+	repo, err := git.PlainInit(mockRepoPath, false)
+	if err != nil {
+		t.Fatalf("Failed to initialize mock git repo: %v", err)
+	}
+
+	readmePath := filepath.Join(mockRepoPath, "README.md")
+	if err := os.WriteFile(readmePath, []byte("# Test Repository"), 0644); err != nil {
+		t.Fatalf("Failed to write file in mock repo: %v", err)
+	}
+
+	worktree, err := repo.Worktree()
+	if err != nil {
+		t.Fatalf("Failed to get worktree: %v", err)
+	}
+	_, err = worktree.Add("README.md")
+	if err != nil {
+		t.Fatalf("Failed to add file to worktree: %v", err)
+	}
+	_, err = worktree.Commit("Initial commit", &git.CommitOptions{
```

Comment:
I see a double commit there, might be good to define a function like `makeTestRepo`